joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/RubyIntermediateAst.scala

```diff
@@ -193,7 +193,10 @@ object RubyIntermediateAst {
   final case class DynamicLiteral(typeFullName: String, expressions: List[RubyNode])(span: TextSpan)
       extends RubyNode(span)
 
-  final case class RangeExpression(lowerBound: RubyNode, upperBound: RubyNode)(span: TextSpan) extends RubyNode(span)
+  final case class RangeExpression(lowerBound: RubyNode, upperBound: RubyNode, rangeOperator: RubyNode)(span: TextSpan)
```

Comment:
If `rangeOperator` is always of type `RangeOperator` then we can do as is below
```suggestion
  final case class RangeExpression(lowerBound: RubyNode, upperBound: RubyNode, rangeOperator: RangeOperator)(span: TextSpan)
```

Reply:
It'll save you from casting the whole time